File: Windows/OpenIoT.App/OpenIoT.Lib/Tools/Persistence/IPersistence.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Tools.Persistence
{
    public interface IPersistence
    {
        void SetString(string key, string value);
        string GetString(string key);
        void SetObject(string key, object value);
        public T GetObject<T>(string key);
        void Delete(string key);
    }
}

File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/tools/persistence/IPersistence.java
package com.palitri.openiot.construction.framework.tools.persistence;

import java.io.Serializable;

public interface IPersistence {
    void SetString(String key, String value);
    String GetString(String key);
    void SetObject(String key, Serializable value);
    Object GetObject(String key);
    void Delete(String key);
}
